src/sync_connection_wrapper.rs

```diff
@@ -0,0 +1,254 @@
+//! This module contains a wrapper type
+//! that provides a [`crate::AsyncConnection`]
+//! implementation for types that implement
+//! [`diesel::connection`]. Using this type
+//! might be useful for the following usecases:
+//!
+//! * using a sync Connection implementation in async context
+//! * using the same code base for async crates needing multiple backends
+
+use crate::{AsyncConnection, SimpleAsyncConnection, TransactionManager};
+use diesel::backend::{Backend, DieselReserveSpecialization};
+use diesel::connection::{
+    Connection, LoadConnection, TransactionManagerStatus, WithMetadataLookup,
+};
+use diesel::query_builder::{
+    AsQuery, CollectedQuery, MoveableBindCollector, QueryBuilder, QueryFragment, QueryId,
+};
+use diesel::row::IntoOwnedRow;
+use diesel::{ConnectionResult, QueryResult};
+use futures_util::future::BoxFuture;
+use futures_util::stream::BoxStream;
+use futures_util::{FutureExt, StreamExt, TryFutureExt};
+use std::marker::PhantomData;
+use std::sync::{Arc, Mutex};
+use tokio::task::JoinError;
+
+fn from_tokio_join_error(join_error: JoinError) -> diesel::result::Error {
+    diesel::result::Error::DatabaseError(
+        diesel::result::DatabaseErrorKind::UnableToSendCommand,
+        Box::new(join_error.to_string()),
+    )
+}
+
+/// A wrapper of a [`diesel::connection::Connection`] usable in async context.
+///
+/// It implements AsyncConnection if [`diesel::connection::Connection`] fullfils requirements:
+/// * it's a [`diesel::connection::LoadConnection`]
+/// * its [`diesel::connection::Connection::Backend`] has a [`diesel::query_builder::BindCollector`] implementing [`diesel::query_builder::MoveableBindCollector`]
+/// * its [`diesel::connection::LoadConnection::Row`] implements [`diesel::row::IntoOwnedRow`]
+pub struct SyncConnectionWrapper<C> {
```

Comment:
It would be great to have an example here similar to what we have for the `AsyncConnectionWrapper`.

We also might want to mention that for now only sqlite is supported.

Reply:
Implemented and handled it in DSL doc examples as well.